### PR TITLE
Add OIDC configuration auto discovery support

### PR DESCRIPTION
**Pull request by omerap12**
### Issue

#2921
### Description
Introduced new functionality to alb.ingress.kubernetes.io/auth-idp-oidc annotation, support for OIDC configuration using discoveryEndpoint to fetch OIDC details dynamically.
Enhanced documentation with instructions and example for using discoveryEndpoint in OIDC configuration.
Created unit tests to validate the implementation and functionality of OIDC Discovery feature.

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ingress/auth_config_builder.go [hunk 1]
```diff
@@ -12,6 +14,10 @@ const (
 	defaultAuthSessionCookieName        = "AWSELBAuthSessionCookie"
 	defaultAuthSessionTimeout           = 604800
 	defaultAuthOnUnauthenticatedRequest = "authenticate"
+	issuerKey                           = "issuer"
```
Comment (oliviassss) [inline pkg/ingress/auth_config_builder.go:17]: maybe add the prefix `default` for all the new vals

**Comment by oliviassss** [comment]
@omerap12, thanks for the contribution, overall lgtm. can you please add the manual test in your PR description? Thanks/

**Comment by omerap12** [comment]
> @omerap12, thanks for the contribution, overall lgtm. can you please add the manual test in your PR description? Thanks/

Done :)

**Comment by omerap12** [comment]
@oliviassss , should I close this?

**Pull request closed without merging**